pyvo/dal/tests/test_tap.py

```diff
@@ -175,11 +175,11 @@ def parameters(self, request, context):
             if 'QUERY' in data:
                 assert data['QUERY'] == 'SELECT TOP 42 * FROM ivoa.obsCore'
                 for param in job.parameters:
-                    if param.id_ == 'query':
+                    if param.id_.lower() == 'query':
```

Comment:
I'm not sure any of these should be handled in the tests itself, instead, add an example that uses different cases, so we in fact test for the fact that the code handles it in a case-insensitive way.

Reply:
The changes to the `MockAsyncTAPServer` have to be done to prevent it from lowercasing all incoming job parameters, thus making it impossible to test this fix. Once it no longer lowercases parameter ids, the other places where it references parameter ids also have to canonicalize. I didn't make that very clear in the PR, now fixed.